lib/puppet/pops/types/types.rb

```diff
@@ -603,6 +671,111 @@ def _assignable?(o)
       end
     end
 
+    class PIterableType < PAnyType
+      attr_reader :element_type
+
+      def initialize(type)
+        @element_type = type
+      end
+
+      def instance?(o)
+        if @element_type.nil? || @element_type.assignable?(PAnyType::DEFAULT)
+          # Any element_type will do
+          case o
+          when Iterable, String, Hash, Enumerable, Enumerator, Range, PEnumType
+            true
+          when Integer
+            o > 0
```

Comment:
Shouldn't this be `>= 0` ? 0.times is just as valid as iterating over an empty array